charon/tests/ui/arrays.out

```diff
@@ -1655,6 +1655,109 @@ fn test_crate::sum_mut_slice<'_0>(@1: &'_0 mut (Slice<u32>)) -> u32
     return
 }
 
+fn test_crate::slice_pattern_1(@1: Array<(), 1 : usize>)
+{
+    let @0: (); // return
+    let x@1: Array<(), 1 : usize>; // arg #1
+    let _named@2: (); // local
+    let @3: (); // anonymous local
+    let @4: &'_ (Array<(), 1 : usize>); // anonymous local
+    let @5: &'_ (()); // anonymous local
+
+    @fake_read(x@1)
+    @4 := &x@1
+    @5 := @ArrayIndexShared<'_, (), 1 : usize>(move (@4), const (0 : usize))
+    _named@2 := copy (*(@5))
+    @3 := ()
+    @0 := move (@3)
+    drop _named@2
+    @0 := ()
+    return
+}
+
+fn test_crate::slice_pattern_2<'_0, T>(@1: Array<&'_0 mut (T), 3 : usize>)
+{
+    let @0: (); // return
+    let x@1: Array<&'_ mut (T), 3 : usize>; // arg #1
+    let _a@2: &'_ mut (T); // local
+    let _b@3: &'_ mut (T); // local
+    let _c@4: &'_ mut (T); // local
+    let @5: (); // anonymous local
+    let @6: &'_ mut (Array<&'_ mut (T), 3 : usize>); // anonymous local
+    let @7: &'_ mut (&'_ mut (T)); // anonymous local
+    let @8: &'_ mut (Array<&'_ mut (T), 3 : usize>); // anonymous local
+    let @9: &'_ mut (&'_ mut (T)); // anonymous local
+    let @10: &'_ mut (Array<&'_ mut (T), 3 : usize>); // anonymous local
+    let @11: &'_ mut (&'_ mut (T)); // anonymous local
+
+    @fake_read(x@1)
+    @10 := &mut x@1
+    @11 := @ArrayIndexMut<'_, &'_ mut (T), 3 : usize>(move (@10), const (0 : usize))
+    _a@2 := move (*(@11))
+    @8 := &mut x@1
+    @9 := @ArrayIndexMut<'_, &'_ mut (T), 3 : usize>(move (@8), const (1 : usize))
+    _b@3 := move (*(@9))
+    @6 := &mut x@1
+    @7 := @ArrayIndexMut<'_, &'_ mut (T), 3 : usize>(move (@6), const (2 : usize))
+    _c@4 := move (*(@7))
+    @5 := ()
+    @0 := move (@5)
+    drop _c@4
+    drop _b@3
+    drop _a@2
+    @0 := ()
+    return
+}
+
+fn test_crate::slice_pattern_3<'_0>(@1: &'_0 (Array<(), 1 : usize>))
+{
+    let @0: (); // return
+    let x@1: &'_ (Array<(), 1 : usize>); // arg #1
+    let _named@2: &'_ (()); // local
+    let @3: (); // anonymous local
+    let @4: &'_ (Array<(), 1 : usize>); // anonymous local
+    let @5: &'_ (()); // anonymous local
+
+    @fake_read(x@1)
+    @4 := &*(x@1)
+    @5 := @ArrayIndexShared<'_, (), 1 : usize>(move (@4), const (0 : usize))
+    _named@2 := &*(@5)
+    @3 := ()
+    @0 := move (@3)
+    drop _named@2
+    @0 := ()
+    return
+}
+
+fn test_crate::slice_pattern_4<'_0>(@1: &'_0 (Slice<()>))
+{
+    let @0: (); // return
+    let x@1: &'_ (Slice<()>); // arg #1
+    let @2: usize; // anonymous local
+    let @3: usize; // anonymous local
+    let @4: bool; // anonymous local
+    let _named@5: &'_ (()); // local
+    let @6: &'_ (Slice<()>); // anonymous local
+    let @7: &'_ (()); // anonymous local
+
+    @fake_read(x@1)
+    @2 := len(*(x@1))
```

Comment:
Here's the `RValue::Len` that caused the CI failure. I guess we just have to parse it correctly, because it's important t have it here.